### PR TITLE
Improve text on private memory page management

### PR DESCRIPTION
**Pull request by ereshetova**
Signed-off-by: Elena Reshetova <elena.reshetova@intel.com>

### CODE REVIEW

**Comment by ereshetova** [comment]
Do not merge yet, need Alyssa to review this.

File: source/security-spec.rst [hunk 1]
```diff
@@ -684,10 +686,27 @@ malicious hypervisor removing a memory page as explained in the above
...
+1. TD guest accesses a private GPA for which the Secure EPT entry is
+in PENDING state and ATTRIBUTES.SEPT\_VE\_DISABLE TD guest attribute is not set. 
+2. TDX module can raise a #VE as a notification mechanism when it detects
+excessive Secure EPT violations raised by the same TD instruction 
+(zero-step attack is detected by TDX module). This is only done if
+bit 0 of TDCS.NOTIFY_ENABLES field is set. 
+
+To avoid the above situations the TD Linux guest kernel verifies that the
+ATTRIBUTES.SEPT\_VE\_DISABLE attribute is set to 1 and bit 0 of
+TDCS.NOTIFY_ENABLES field is not set during its initialization process.
```
Comment (noopwafel) [inline source/security-spec.rst]: just to be clearer: "To ensure the above situations do not occur, the TD Linux guest kernel requires that ATTRIBUTES.SEPT\_VE\_DISABLE is set, and that bit 0 of TDCS.NOTIFY_ENABLES is not set. These values are checked during boot."?

File: source/security-spec.rst [hunk 2]
```diff
@@ -684,10 +686,27 @@ malicious hypervisor removing a memory page as explained in the above
...
+
+Note that such behavior does have an implication for the zero-step attack mitigation
+for the TD guest, i.e. Linux TD guest kernel does not
+employ its own protection mechanisms against this type of attack beyond the
+protections implemented in the TDX module.
```
Comment (noopwafel) [inline source/security-spec.rst]: "Although this disables TDX module notifications for excessive numbers of Secure EPT violations, the basic defenses against zero-stepping provided by the TDX module are still in effect."?

File: source/security-spec.rst [hunk 3]
```diff
@@ -627,15 +627,17 @@ TDX guest private memory page management
...
-module, TDX guest memory access to a non-accepted page results in a #VE
-inserted by the TDX guest module. It is very important for security that
-it does not happen during certain TDX guest critical code paths (see
-`Safety against #VE in kernel code`_ for more details).
+module (and given that a TD guest opts-in for such notifications), certain events
+(like TDX guest memory access to a non-accepted page) can result in a #VE
+inserted by the TDX guest module. Please see section 16.3.3 in
+`Intel TDX module architecture specification <https://www.intel.com/content/dam/develop/external/us/en/documents/tdx-module-1.0-public-spec-v0.931.pdf>`_ for more details. For the Linux kernel running as
+a TDX guest is very important that such #VE notifications do not happen during certain TDX
+guest critical code paths (see `Safety against #VE in kernel code`_ for more details).
```
Comment (noopwafel) [inline source/security-spec.rst:640]: maybe just "For the Linux kernel" without "running as a TDX guest"?

**Review by noopwafel — APPROVED** [review]
LGTM, I suggested some language improvements in comments but I think fine as-is too

File: source/security-spec.rst (hunk 1 above)
Comment (ereshetova) [reply source/security-spec.rst:702]: Yes, they will be checked during TD initialization during TD guest boot.

**Pull request merged**